integration/spark-common/src/main/scala/org/apache/spark/rdd/CarbonMergeFilesRDD.scala

```diff
@@ -94,6 +102,28 @@ object CarbonMergeFilesRDD {
           }
       }
     }
+    if (carbonTable.isHivePartitionTable) {
+      segmentIds.foreach(segmentId => {
+        val readPath: String = CarbonTablePath.getSegmentFilesLocation(tablePath) +
+                               CarbonCommonConstants.FILE_SEPARATOR + segmentId + "_" +
+                               segmentFileNameToSegmentIdMap.get(segmentId) + ".tmp"
+        // Merge all partition files into a single file.
+        val segmentFileName: String = SegmentFileStore
+          .genSegmentFileName(segmentId, segmentFileNameToSegmentIdMap.get(segmentId))
+        val segmentFile: SegmentFileStore.SegmentFile = SegmentFileStore
+          .mergeSegmentFiles(readPath,
+            segmentFileName,
+            CarbonTablePath.getSegmentFilesLocation(tablePath))
+
+        if (segmentFile != null) {
```

Comment:
I think this code is not required for partition table as already segments file written to segment folder